homeassistant/components/bunq/sensor.py

```diff
@@ -0,0 +1,153 @@
+
+    # schedule updates for sensors
+    data = BunqData(hass, api_context, sensors)
+    await data.schedule_update(UPDATE_INTERVAL)
+
+
+def get_account_data():
+    """Get active bunq accounts."""
+    active_accounts = []
+
+    accounts = endpoint.MonetaryAccountBank.list().value
+    for account in accounts:
+        if account.status == "ACTIVE":
+            active_accounts.append(account)
+
+    return active_accounts
+
+
+class BunqBalanceSensor(Entity):
+    """Setup bunq balance sensor."""
+
+    def __init__(self, account):
+        """Initialize the sensor."""
+        self.id = account.id_
+        self._name = account.description
+        self._state = float(account.balance.value)
+        self._unit_of_measurement = account.currency
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the entity icon."""
+        return ICON
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    def load_data(self, data):
+        """Update sensor data."""
+        try:
+            old_balance = self._state
+            self._state = data.get(self.id)
+            return self._state == old_balance
+        except KeyError:
+            self._state = None
+            _LOGGER.warning("Count not find key in API results.")
+            return False
+
+
+class BunqData:
+    """Get the latest data and updates the sensors."""
+
+    def __init__(self, hass, api_context, sensors):
+        """Initialize the data object."""
+        self._api_context = api_context
+        self._sensors = sensors
+        self.data = {}
+        self.hass = hass
+
+    async def update_devices(self):
+        """Update all sensors."""
+        tasks = []
+
+        for sensor in self._sensors:
+            if sensor.load_data(self.data):
+                tasks.append(sensor.async_update_ha_state())
+        if tasks:
+            await asyncio.wait(tasks)
+
+    async def schedule_update(self, seconds):
+        """Schedule an update."""
+        async_track_time_interval(
+            self.hass, self.async_update, timedelta(seconds=seconds)
+        )
+
+    async def async_update(self, *_):
+        """Update data."""
+        try:
+            # get new data from api
+            accounts = get_account_data()
+
+            # create a dict with account id as key and account data as value
+            self.data = {
+                account.id_: float(account.balance.value) for account in accounts
+            }
+
+            # update the sensors
+            await self.update_devices()
+        except KeyError:
+            _LOGGER.warning("Bunq account not found.")
```

Comment:
Do not include the name of the integration in your log message. See [style guidelines](https://developers.home-assistant.io/docs/en/development_guidelines.html#log-messages) for reference.
```suggestion
            _LOGGER.warning("Account not found")
```